runtime-light/stdlib/string/regex-functions.cpp

```diff
@@ -102,6 +107,109 @@ int64_t skip_utf8_subsequent_bytes(int64_t offset, const std::string_view subjec
   return offset;
 }
 
+std::optional<backref> try_get_backref(std::string_view preg_replacement) noexcept {
+  if (preg_replacement.empty() || preg_replacement[0] < '0' || preg_replacement[0] > '9') {
```

Comment:
Maybe do it like `if (preg_replacement.empty() || !std::isdigit(preg_replacement[0]))`?